ddp-workspace/projects/ddp-singular/src/app/app.component.ts

```diff
@@ -1,18 +1,61 @@
-import { Component, ElementRef } from '@angular/core';
+import { Component, ElementRef, HostListener } from '@angular/core';
+import { AnalyticsEventsService } from 'ddp-sdk';
+
+import { GTagEvent } from './constants/gtag-event';
+import { Route } from './constants/route';
+import { IGNORE_ANALYTICS_CLASS } from './constants/analytics';
 
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
-export class AppComponent  {
+export class AppComponent {
   title = 'ddp-singular';
 
-  constructor(private elRef: ElementRef) {
-  }
+  constructor(
+    private elRef: ElementRef,
+    private analytics: AnalyticsEventsService
+  ) {}
 
   onActivate(): void {
     this.elRef.nativeElement.scrollTo(0,0);
   }
+
+  // based on https://plausible.io/blog/track-outbound-link-clicks#:~:text=%3C-,script,-%3E%0A%20%20document
+  @HostListener('document:click', ['$event'])
+  monitorOutboundLinks(event: Event): void {
+    const link = this.findNearestAnchorElement(event.target);
+
+    // stop if it's the same origin navigation
+    if (!link || !link.href || !link.host || link.host === location.host || link.classList.contains(IGNORE_ANALYTICS_CLASS)) {
```

Comment:
**_#minor_** 

the condition can be extracted as a separate method that returns boolean,
e.g. `shouldNotSendAnalytics: () => boolean`

Reply:
Done:  9d6ed676bfbbde332953a9e6cd81a63b54ce5c17

Reply:
@grigol-brd I think this is ready. If you or @alextanlan approve it, I think you can just merge it yourself. You should just that as I think they are quite anxious to make this available to users.